design/worker-config-drive.md

```diff
@@ -0,0 +1,178 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# worker-config-drive
+
+## Status
+
+provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [Title](#title)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories [optional]](#user-stories-optional)
+            * [Story 1](#story-1)
+            * [Story 2](#story-2)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+      * [References](#references)
+
+<!-- Added by: stack, at: 2019-02-15T11:41-05:00 -->
+
+<!--te-->
+
+## Summary
+
+In order to bring a host into the cluster as a worker we have to give
+it the correct ignition configuration as it boots. The contents of the
+Ignition file are available via the kubernetes API. Something needs to
+build a config drive image containing the ignition file and the
+baremetal operator needs to pass that formatted image to Ironic as
+part of provisioning the host.
+
+In order to avoid having the abstraction around Ironic break down,
+something that knows about the expectations of the image being
+deployed (especially the path where the Ignition file should be
+written within the config drive image) should build the drive image,
+and pass it to the baremetal operator via the host along with the
+other provisioning instructions like the image name and checksum.
+
+The Ignition file contents are stored in a Secret within the
+kubernetes database because they contain certificate authority
+information. We will need to store the config drive contents in a
+secret as well, to avoid leaking the data when we pass the drive image
+to the baremetal operator.
```

Comment:
This feels like an implementation detail. Does it impact the overall behavior that can be leveraged?

Reply:
If the operator doesn't build the config drive image, it has to be given the contents securely.